src/components/button/index.tsx

```diff
@@ -1,18 +1,30 @@
 import React, { FC, MouseEvent, ReactNode } from 'react';
 import { Button as ChakraButton } from '@chakra-ui/react';
 
-interface Props {
+type SharedProps = {
   variant?: 'primary' | 'secondary';
   size?: 'md' | 'lg';
   isDisabled?: boolean;
-  onClick: (event: MouseEvent<HTMLElement>) => void;
   children: ReactNode;
-}
+};
+
+type ButtonProps = SharedProps & {
+  type?: 'button';
+  onClick: (event: MouseEvent<HTMLElement>) => void;
+};
+
+type SubmitProps = SharedProps & {
```

Comment:
Why do you need to split the props and cannot just `type?: 'button | 'submit'`?

Reply:
Because `onClick` should be mandatory for the `button`, but optional for `submit`